.github/workflows/ci-build.yaml

```
@@ -57,9 +57,74 @@ env:
   TESTCONTAINERS_RYUK_DISABLED: true
 
 jobs:
-  initial-mvn-install:
+  pre-build-checks:
     if: github.repository == 'apache/camel-quarkus'
     runs-on: ubuntu-latest
+    outputs:
+      continue-build: ${{ steps.pre-build-checks.outputs.continue-build }}
+      run-checks: ${{ steps.init.outputs.run-checks }}
+    steps:
+      - name: Initialize
+        id: init
+        run: |
+          if [[ "${{ github.event_name }}" == "pull_request" ]] && [[ "${{ github.actor }}" == "dependabot[bot]" ]]; then
+            echo "run-checks=true" >> $GITHUB_OUTPUT
+          else
+            echo "run-checks=false" >> $GITHUB_OUTPUT
+          fi
+      - name: Set up JDK 17
+        uses: actions/setup-java@v2
+        if: steps.init.outputs.run-checks == 'true'
+        with:
+          distribution: 'temurin'
+          java-version: '17'
+      - name: Checkout
+        uses: actions/checkout@v2
+        if: steps.init.outputs.run-checks == 'true'
+        with:
+          ref: ${{ github.head_ref }}
+          fetch-depth: 0
+      - name: Pre build checks
+        id: pre-build-checks
+        run: |
+          if [[ "${{ steps.init.outputs.run-checks }}" == "true" ]]; then
+            ./mvnw cq:sync-versions -N ${CQ_MAVEN_ARGS}
+            ./mvnw clean install -N ${CQ_MAVEN_ARGS}
+
+            cd poms
+            ../mvnw clean install -N ${CQ_MAVEN_ARGS}
+
+            cd bom
+            ../../mvnw process-resources -Dcq.flatten-bom.format --fail-never ${CQ_MAVEN_ARGS}
+            ../../mvnw clean install ${CQ_MAVEN_ARGS}
+
+            cd ../../
```

A similar result could perhaps be achieved with less lines:
```suggestion
            ./mvnw clean install -f poms/bom -am -Dcq.flatten-bom.format --fail-never ${CQ_MAVEN_ARGS}
            # the above may change the BOM in the source tree, but does not install it, so let's install it
            ./mvnw clean install ${CQ_MAVEN_ARGS} -f poms/bom
```

---

Thanks! I'll give that a test.

---

The flatten-bom default to process-resources phase and is not installed in the first "mvn clean install" line ? If someone has light to share, that would be welcome.

Also, if second line is needed, maybe `mvnw install` without `clean` could be enough ?

---

> The flatten-bom default to process-resources phase and is not installed in the first "mvn clean install" line ? If someone has light to share, that would be welcome.

I think the suggestion made [here](https://github.com/jamesnetherton/camel-quarkus/commit/281c5789e68733dfcbf8e16da9c24adfc15b8b1e#r100230924), was that we want the relevant POM hierarchy installed before `cq.flatten-bom.format` runs. Thus `mvnw clean install -pl poms/bom -am -Dcq.flatten-bom.format` would be enough to do everything we need in one step.

---

I have reduced the number of required `mvnw` steps as suggested.

---

Ok, so in such a situation. It's really needed to do it in 2 steps. Many thanks for sharing :+1:

---

> The flatten-bom default to process-resources phase and is not installed in the first "mvn clean install" line ?

Yes, and I know no way how to do it better. The problem is that `-Dcq.flatten-bom.format` may add some exclusions and other edits to the source `pom.xml` that the further flattening does not see, because it uses Maven APIs to get the entries of the flattened POM. Maven APIs read the source pom.xml file only once before I do the edits. 
So currently, if `-Dcq.flatten-bom.format` does some edits, it throws an exception that recommends the user to invoke `mnv install` again. `--fail-never` is thus need so that it does not fail the whole CI job.

---

Thanks for sharing the details @ppalaga. Indeed, this is a tricky situation and no better idea than a 2 pass treatment from my side. That's great to have those flatten boms.